src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java

```diff
@@ -139,4 +141,24 @@ private RetrieveUserProfilesRequest generateRetrieveUserProfilesRequest(Organisa
     public ProfessionalUser persistUser(ProfessionalUser updatedProfessionalUser) {
         return professionalUserRepository.save(updatedProfessionalUser);
     }
+
+    @Override
+    public UserRolesResponse modifyRolesForUser(ModifyUserProfileData modifyUserProfileData, String userId) {
+        UserRolesResponse userRolesResponse;
+        try (Response response =  userProfileFeignClient.modifyUserRoles(modifyUserProfileData, userId)) {
+
+            Class clazz = UserRolesResponse.class;
+            ResponseEntity responseResponseEntity = JsonFeignResponseHelper.toResponseEntity(response, clazz);
+
+            if (response.status() > 300) {

```

Comment:
Please avoid nesting. There are too many logical branches here. Instead you can extract these blocks out to private methods or make the code more generic and use a utility method with a public method to perform these operations. Lambda expressions are also an option wherever you can provide a functional interface.
Please refactor.